nbconvert/preprocessors/clearmetadata.py

```diff
@@ -3,20 +3,47 @@
 # Copyright (c) IPython Development Team.
 # Distributed under the terms of the Modified BSD License.
 
-from traitlets import Set
+from traitlets import Bool, Set
 from .base import Preprocessor
 
 class ClearMetadataPreprocessor(Preprocessor):
     """
     Removes all the metadata from all code cells in a notebook.
     """
 
+    clear_notebook_metadata = Bool(True,
```

Comment:
Do we perhaps want the ability to choose between notebook metadata, cell metadata or both? It might be easier than specifically setting every piece of cell metadata that we want to keep.